crypto/asn1/bio_asn1.c

```diff
@@ -138,6 +138,13 @@ static int asn1_bio_free(BIO *b)
     if (ctx == NULL)
         return 0;
 
+    if (ctx->prefix_free) {
```

Comment:
This looks like a double free here?
ctx->prefix_free is not set to NULL after the cleanup().

Reply:
The implementation ndef_prefix_free() seems to be safe to be called multiple times. It is freeing "ndef_aux->derbuf" and setting the pointer to NULL afterwards. Calling OPENSSL_free() with NULL should be safe as far as I know.

Alternative would be to zero the free function ptr after it was called. It looks like the state machine cannot be reset to the beginning.

What do you suggest?